src/containers/Earn/index.tsx

```diff
@@ -17,24 +22,42 @@ const Earn = () => {
     const { nitroPoolsModel: nitroPoolsActions } = useStoreActions((state) => state)
     const { nitroPools } = nitroPoolsState
 
-    useEffect(() => {
-        if (!geb) return
-        async function fetchPools() {
-            for (const pool of POOLS) {
-                try {
-                    await nitroPoolsActions.fetchNitroPool({
-                        userAddress: account ?? undefined,
-                        camelotPoolAddress: pool.camelotPoolAddress,
-                        nitroPoolAddress: pool.nitroPoolAddress,
-                        geb,
+    const [cachedPools, setCachedPools] = useState<Cache>({})
+
+    const fetchPools = useMemo(() => {
+        if (!geb) return () => {}
+        return async () => {
+            try {
+                const poolResults = await Promise.all(
+                    POOLS.map(async (pool: { camelotPoolAddress: any; nitroPoolAddress: any }) => {
```

Comment:
Should this be moved to the usePools hook?

Reply:
I tried adding it to the hook but it didn't make a difference because it's not used in multiple components so there's no performance improvement. Best to just memoize until we use it in multiple places